 .
 samba (2:4.8.0-6.netgear2) unstable; urgency=medium
 .
   * Fixed an issue with invalid queries for
     uppercase input.
Author: Wenbo Guo <wenbo.guo@netgear.com>

---

--- samba-4.8.0.orig/source3/rpc_server/mdssvc/sparql_parser.y
+++ samba-4.8.0/source3/rpc_server/mdssvc/sparql_parser.y
@@ -373,7 +373,7 @@ static const char *map_expr(const char *
 		}
 		sparql = talloc_asprintf(talloc_tos(),
 					 "?obj %s ?%c "
-					 "FILTER(fn:contains(fn:lower-case(?%c), '%s'))",
+					 "FILTER(regex(?%c, '.*%s.*', 'i'))",
 					 p->sparql_attr,
 					 s->var,
 					 s->var,
@@ -484,7 +484,7 @@ char *replace_umlaut(const char* src)
 	char *des = (char*)malloc(path_len + 1);
 	if (NULL == des)
 	{
-		return NULL;	
+		return NULL;
 	}
 
 	strncpy(des, src, path_len);
